e2e/utils.ts

```diff
@@ -279,3 +288,17 @@ export const prepareStandaloneSetup = (fixtureName: string) => {
   };
   return startApp;
 };
+
+export async function waitForHydration(page: Page) {
+  await page.waitForFunction(
+    () => {
+      const el = document.querySelector('body');
+      if (el) {
+        const keys = Object.keys(el);
+        return keys.some((key) => key.startsWith('__reactFiber'));
+      }
+    },
+    null,
+    { timeout: 3000 },
+  );
+}
```

Comment:
I added this utility and this seems necessary in a few client interaction specific tests. I'm not sure why this wasn't the issue in the current version and there might be a crucial change how hydration kicks in. This is probably worth investigating.

Reply:
I agree. It feels unnecessary, ideally, but I'm not sure at the moment.